engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -17,9 +18,40 @@ def self.update_cache(remote_ip, system_login, product_id, registry: false)
     )
   end
 
+  def self.update_cache_not_payg(cache_key, mode)
+    cache_path = get_cache_path(mode)
+    InstanceVerification.write_cache_file(cache_path, cache_key)
+  end
+
   def self.write_cache_file(cache_dir, cache_key)
     FileUtils.mkdir_p(cache_dir)
     FileUtils.touch(File.join(cache_dir, cache_key))
+    Rails.logger.info "#{cache_dir} updated for #{cache_key}"
+  end
+
+  def self.get_cache_path(mode)
+    if mode == 'byos'
+      Rails.application.config.repo_byos_cache_dir
+    elsif mode == 'hybrid'
+      Rails.application.config.repo_hybrid_cache_dir
+    else
+      # payg
+      Rails.application.config.repo_payg_cache_dir
+    end
+  end
+
+  def self.get_cache_entries(mode)
+    cache_path = InstanceVerification.get_cache_path(mode)
+    FileUtils.mkdir_p(cache_dir)
+    Dir.children(cache_path)
+  end
+
+  def self.reg_code_in_cache?(reg_code, product_hash, mode)
```

Comment:
Generic method name but specific implementation to BYOS.

Reply:
`BYOS` and `HYBRID`, I understand based on the comments we want to include `PAYG`